server/models/Document.ts

```diff
@@ -620,14 +620,6 @@ class Document extends ParanoidModel {
       collectionIds = await user.collectionIds();
     }
 
-    // If the user has access to no collections then shortcircuit the rest of this
-    if (!collectionIds.length) {
```

Comment:
Probably retain but with the addition of `&& !options.includeDrafts`?

Reply:
Well, not needed anymore. If `!collectionIds.length && !options.includeDrafts` evaluates to `true`, then the `whereClause` further down evaluates to:
```sql
"searchVector" @@ to_tsquery('english', :query) AND
"teamId" = :teamId AND
"collectionId" IS NULL AND "createdById" = :userId AND
"archivedAt" IS NULL AND
"publishedAt" IS NOT NULL AND
"deletedAt" IS NULL
```
And we know, by design that if `"collectionId" IS NULL`, then the document must be an unpublished one.